test_conformance/conversions/basic_test_conversions.h

```diff
@@ -232,6 +245,7 @@ struct TestType
             case kshort: return std::is_same<cl_short, T>::value;
             case kuint: return std::is_same<cl_uint, T>::value;
             case kint: return std::is_same<cl_int, T>::value;
+            case khalf: return std::is_same<cl_half, T>::value;
```

Comment:
Note: since a `cl_half` and a `cl_ushort` are the same type this function doesn't quite behave as expected in all cases after this change.  It would be really nice if we could differentiate between these two types.

Reply:
Correct, please notice `isTypeFp` array which takes care of distinguishing `cl_half` and `cl_ushort` later. I agree it looks a bit hard coded but it may be difficult to figure out more elegant solution in this case.

Reply:
This is fixed now, by also looking at the FP template parameter.  Resolving.

```c++
case khalf: return std::is_same<cl_half, T>::value && FP;
```